tasks/image_configuration/set-fw-env-config/Kconfig

```diff
@@ -0,0 +1,7 @@
+config DS_GENERATE_FW_ENV_CONFIG
+	default y
```

Comment:
If there isn't a config to match the model's compatible string, an error message is printed and comments put in `/etc/fw_env.config` explaining what needs to be added where:
https://github.com/embeddedTS/distro-seed/pull/51/files/9a8b20f13611d21b18734f11f9d4ff281b3f83f2#diff-63a030d709a8ff35ffdbabb411dce48fb6c7ad0e8a6d94f1087eae95f2036e74R57-R60

Reply:
Yes, that's my point, e.g.:
```
[FAILED] Failed to start fw_env.ser…k to an appropriate fw_env.config.
See 'systemctl status fw_env.service' for details.
```

On a stock Deb 12 Headless build booted on a TS-7970.

Either, every board that we support with `distro-seed` should have a compatible `fw_env.conf` set up and installed as a part of this PR. Or, change this to not a default option and is enabled on platforms as needed/used.

Otherwise, we're shipping an init script that shows a big red "FAILED" on the majority of platforms that `distro-seed` supports.

Reply:
You're absolutely right. I'll rework the behavior when no config file is found (and/or when no valid environment is found despite config file(s) being present). I was hoping to leave behind something useful if the user has to investigate/address why the `fw_*env` commands are failing.

Reply:
The script now returns 0, not 1, if it can't find any matching configs.

It still will place a default, comment-only file in `/run/fw_env.config` to make what-to-do-about-it easier when `fw_printenv` fails.

Reply:
I'll defer to @markfeathers on this for a final decision, but, I disagree with the script now succeeding on failure. This means failures would be hidden on systems that are supported but something went wrong.

I stand by my original statement. This should not be default if there are platforms in `distro-seed` that are unsupported, but can be default if this PR adds support for all current platforms. I'm fine with enforcing that new platforms either _need_ this support as part of their BSP or it needs to be explicitly disabled in the defconfig; that makes sense to me. If you want to update this PR to touch defconfigs that are unsupported and disable this configuration in them, I would accept that too.

But in order for me to sign off on this PR, with this PR added we should have nothing generating a failure at boot, and shouldn't be hiding failures on boot to work around it.

Reply:
I'm not entirely sure what you mean, since we already probe devices and start services for things not supported on a platform. Are those considered failures being hidden? And there's GPSD coming right behind this PR, which will have a somewhat similar set of problems.  This may be a justification for having platform-specific defconfigs, but at the moment we are still trying to keep these defconfigs "as common as possible".

All platforms previously supported by distro-seed remain supported. If we have additional platforms, I invite you and everyone to review the compatible strings under `tasks/components/ts-uboot-env-configs/files` to see what platforms need to be added.

If no valid config is found, then in addition to `/etc/fw_env.config` having instructions on what to do, the output of `systemctl status fw-env` also contains a message saying what happened. If the `u-boot-tools` package is installed, this is way kinder than doing nothing at boot.

Another idea to resolve this would be to run the check run at first boot, and then have the service disable itself if no file is found for the booted platform.

Reply:
> I'm not entirely sure what you mean, since we already probe devices and start services for things not supported on a platform. Are those considered failures being hidden?

Can you provide an example of this?
It largely depends on the service. For example, a platform that doesn't support IIO starts `iiod`. The starting of iiod succeeded and that's what the message indicates. Whether or not the platform has IIO is irrelevant in this case. In the case of a script copying files to support a feature, if those files don't exist and that feature will not work, that should be an error. If `iiod` required a config file that is not present, then `iiod` would fail, and systemd would report the failure.


> All platforms previously supported by distro-seed remain supported. If we have additional platforms, I invite you and everyone to review the compatible strings under `tasks/components/ts-uboot-env-configs/files` to see what platforms need to be added.

Platforms supported by `distro-seed` that would be affected by the config option being set to default but not having `fw_env.conf` files (after your latest round of changes):
TS-4100
TS-7400-V2
TS-7670
TS-TPC-7990


> If no valid config is found, then in addition to `/etc/fw_env.config` having instructions on what to do, the output of `systemctl status fw-env` also contains a message saying what happened. If the `u-boot-tools` package is installed, this is way kinder than doing nothing at boot.

That is fine, but, if no valid config is found, then the service should error on startup. The issue comes when this gets enabled by default on platforms that don't have configs so we are shipping errors.


> Another idea to resolve this would be to run the check run at first boot, and then have the service disable itself if no file is found for the booted platform.

In the case of this script and this feature, a failure should always fail. Because it would mean that files that should exist, do not.

Reply:
It looks like this is moot point as the config option that this originally implemented is no longer being used, and the config option to support it now is not enabled by default.